libqpdf/QPDF.cc

```diff
@@ -2346,9 +2346,8 @@ QPDF::reserveObjects(QPDFObjectHandle foreign, ObjCopier& obj_copier, bool top)
 
     if (foreign_tc == ::ot_array) {
         QTC::TC("qpdf", "QPDF reserve array");
-        int n = foreign.getArrayNItems();
-        for (int i = 0; i < n; ++i) {
-            reserveObjects(foreign.getArrayItem(i), obj_copier, false);
+        for (auto const& item: foreign.as_array()) {
```

Comment:
It is excellent that you have split out this and the corresponding one for dictionaries into separate commits. This will be very helpful when I try to upload into my brain how to make use of the new APIs.

I wonder whether we should have a preprocessor symbol that, when defined, adds a warning around uses of aitems and ditems to help people locate them. Probably not...too complex. Better to have people just search.

Reply:
I was considering this too, but slightly more widely  than aitems / ditems.  Something for 12.1 I think.